From: Vaclav Ovsik <vaclav.ovsik@i.cz>
Subject: [PATCH] fixes/nonlinux

Removes conditional compilation directives `#ifdef __linux__',
so the build process can succeed on non linux (Debian GNU/kFreeBSD).
Solves the bug http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=536890

Signed-off-by: Vaclav Ovsik <vaclav.ovsik@i.cz>

---
 T/ctst_11_io.c    | 4 ----
 T/ctst_3_medium.c | 2 --
 T/ctst_99_64bit.c | 4 ----
 3 files changed, 10 deletions(-)

diff --git a/T/ctst_11_io.c b/T/ctst_11_io.c
index 5b8e32e..201e2e6 100644
--- a/T/ctst_11_io.c
+++ b/T/ctst_11_io.c
@@ -274,14 +274,10 @@ int tst(void)
 
   errno = 0;
   ASSERT(!ustr_io_putfilename(&s2, "doesn't exist.txt", "rb"));
-#ifdef __linux__
   ASSERT(errno == ENOENT);
-#endif
   errno = 0;
   ASSERT(!ustr_io_putfilename(&s2, "./doesn't exist/foo.txt", "rb"));
-#ifdef __linux__
   ASSERT(errno == ENOENT);
-#endif
   ASSERT(ustr_cmp_buf_eq(s2, ustrp_cstr(sp1), ustrp_len(sp1)));
 
 #ifdef ENOSPC
diff --git a/T/ctst_3_medium.c b/T/ctst_3_medium.c
index d32414f..0d3d02f 100644
--- a/T/ctst_3_medium.c
+++ b/T/ctst_3_medium.c
@@ -184,7 +184,6 @@ int tst(void)
   ASSERT( ustr_setf_share(s2));
   ASSERT( ustr_setf_owner(s2));
   ASSERT( ustr_add(&s2, USTR("")));
-#ifdef __linux__
   {
     wchar_t *wt = (wchar_t *)"\xFF\xFF\xFF\xFF";
   /* test multi-byte conversion failure */
@@ -196,7 +195,6 @@ int tst(void)
   ASSERT(!ustr_sc_sub_fmt(&s2, 1, 2, "%ls", wt));
   }
 #endif
-#endif
   ustr_sc_free2(&s2, ustr_dupx(1, 0, 0, 0, s2));
   
   ASSERT(!ustr_exact(s2));
diff --git a/T/ctst_99_64bit.c b/T/ctst_99_64bit.c
index aa084b6..8b15d58 100644
--- a/T/ctst_99_64bit.c
+++ b/T/ctst_99_64bit.c
@@ -2,15 +2,12 @@
 
 static const char *rf = __FILE__;
 
-#ifdef __linux__
 #include <unistd.h>
 #include <sys/types.h>
 #include <sys/mman.h>
-#endif
 
 static int big_tst(size_t fsz, size_t usz, int hack, int tst_add)
 {
-#ifdef __linux__
   int fd = -1;
   void *ptr = NULL;
   Ustr *s3 = NULL;
@@ -83,7 +80,6 @@ static int big_tst(size_t fsz, size_t usz, int hack, int tst_add)
   close(fd);
  fail_mkstemp:
   return (EXIT_FAILED_OK);
-#endif
 }
 
 int tst(void)
-- 
tg: (b148fb7..) fixes/nonlinux (depends on: upstream)
